src/v2i-hub/CARMAStreetsPlugin/test/test_JsonToJ3224SDSMConverter.cpp

```diff
@@ -20,169 +20,679 @@ namespace CARMAStreetsPlugin
+                        "measurement_time": 0,
+                        "time_confidence": 0,
+                        "pos": {
+                            "offset_x": -11,
+                            "offset_y": -20,
+                            "offset_z": -32
+                        },
+                        "pos_confidence": {
+                            "pos": 0,
+                            "elevation": 0
+                        },
+                        "speed": 70,
+                        "speed_confidence": 0,
+                        "speed_z": 50,
+                        "heading": 0,
+                        "heading_conf": 0
+                        },
+                        "detected_object_optional_data": {
+                        "detected_obstacle_data": {
+                            "obst_size": {
+                            "width": 5,
+                            "length": 20,
+                            "height": 10
+                            },
+                            "obst_size_confidence": {
+                            "width_confidence": 0,
+                            "length_confidence": 0
+                            }
+                        }
+                        }
+                    }
+                    }
+                ]
+                }
+        )";
         Json::Value root;
         bool result = converter.parseJsonString(valid_json_str, root);
-        ASSERT_TRUE(result);
+        EXPECT_TRUE(result);
         auto sdsmPtr = std::make_shared<SensorDataSharingMessage>();
         converter.convertJsonToSDSM(root, sdsmPtr);
 
-        ASSERT_EQ(400, sdsmPtr->objects.list.array[0]->detObjOptData->choice.detObst.obstSize.width);
-        ASSERT_EQ(300, sdsmPtr->objects.list.array[0]->detObjOptData->choice.detObst.obstSize.length);
-        ASSERT_EQ(100, *sdsmPtr->objects.list.array[0]->detObjOptData->choice.detObst.obstSize.height);
+        tmx::messages::SdsmEncodedMessage encodedSdsm;
+        converter.encodeSDSM(sdsmPtr, encodedSdsm);
+        EXPECT_EQ(41,  encodedSdsm.get_msgId());
+	
+        std::string expectedSDSMEncHex = "00293400313233343fdf9f2330dd90da406b49d200d693a3fe00000000014011800057700bffa3ff5bfef80011800c80000a0282805000";
+        EXPECT_EQ(expectedSDSMEncHex, encodedSdsm.get_payload_str());
+    }
 
-        ASSERT_EQ(6, sdsmPtr->objects.list.array[0]->detObjOptData->choice.detObst.obstSizeConfidence.widthConfidence);
-        ASSERT_EQ(7, sdsmPtr->objects.list.array[0]->detObjOptData->choice.detObst.obstSizeConfidence.lengthConfidence);
-        ASSERT_EQ(8, *sdsmPtr->objects.list.array[0]->detObjOptData->choice.detObst.obstSizeConfidence.heightConfidence);
 
-    }
+// Test for SDSM optional data - obstacle data
+    TEST_F(test_JsonToJ3224SDSMConverter, convertJsonToSDSM_multiple_objects)
+    {
+        JsonToJ3224SDSMConverter converter;
+        std::string valid_json_str = R"(
+            {
+                "msg_cnt": 128,
```

Comment:
The encoding error was this msg_cnt value out of range. Definition of this refer to: https://github.com/usdot-fhwa-stol/CARMASensitive/blob/b63ca0c26a68680ef575d2c995e2620d5068b006/J2735_201603_2023-06-22.asn#L3938

Reply:
Yeah figured that out too.

Reply:
That was not the cause of the initial errors though since it was failing with message count of 3